yapapi/strategy.py

```diff
@@ -21,10 +23,130 @@
 SCORE_REJECTED: Final[float] = -1.0
 SCORE_TRUSTED: Final[float] = 100.0
 
+PROP_DEBIT_NOTE_INTERVAL_SEC: Final[str] = "golem.com.scheme.payu.debit-note.interval-sec?"
+PROP_PAYMENT_TIMEOUT_SEC: Final[str] = "golem.com.scheme.payu.payment-timeout-sec?"
+PROP_DEBIT_NOTE_ACCEPTANCE_TIMEOUT: Final[str] = "golem.com.payment.debit-notes.accept-timeout?"
+
+MID_AGREEMENT_PAYMENTS_PROPS = [PROP_DEBIT_NOTE_INTERVAL_SEC, PROP_PAYMENT_TIMEOUT_SEC]
+
+
+@dataclass
+class PropValueRange:
+    min: Optional[Union[int, float]] = None
+    max: Optional[Union[int, float]] = None
+
+    def __contains__(self, item: Union[int, float]) -> bool:
+        return (self.min is None or item >= self.min) and (self.max is None or item <= self.max)
+
+    def closest_acceptable(self, item: Union[int, float]):
```

Comment:
PEP 484 Type Hints (https://www.python.org/dev/peps/pep-0484/) says:
> when an argument is annotated as having type float, an argument of type int is acceptable; similar, for an argument annotated as having type complex, arguments of type float or int are acceptable.
```suggestion
    min: Optional[float] = None
    max: Optional[float] = None

    def __contains__(self, item: float) -> bool:
        return (self.min is None or item >= self.min) and (self.max is None or item <= self.max)

    def closest_acceptable(self, item: float):
```

Reply:
"explicit is better than implicit"

Reply:
Imho if there's a PEP saying that `int` is a `float` in this context this is not explicit in a useful way.
You write `1.1 + 7` instead of explicit `1.1 + float(7)`.

Reply:
IOW, implicity accepted in PEP is the OK level of implicity, and "Simple is better than complex."

Reply:
sigh